tutorial_files/scm_read_obs.py

```diff
@@ -0,0 +1,341 @@
+
+  return_dict = {'year': obs_year, 'month': obs_month, 'day': obs_day, 'hour': obs_hour,
+    'time': obs_time, 'date': obs_date, 'time_slice_indices': obs_time_slice_indices,
+    'pres_l': obs_pres_l, 'cld': obs_cld, 'T': obs_T, 'q': obs_q, 'u': obs_u, 'v': obs_v,
+    'pwat': obs_pwat, 'time_h': obs_time_h,
+    'tprcp_rate_accum': obs_precip, 'qv': obs_q, 'rh': obs_rh, 'rh_500': obs_rh_500,
+    'lw_up_TOA_tot': obs_lw_net_toa, 'rad_net_srf': obs_rad_net_srf, 'sw_dn_TOA_tot': obs_sw_dn_toa,
+    'lw_dn_sfc_tot': obs_lw_dn_srf, 'sw_dn_sfc_tot': obs_sw_dn_srf, 'lwp': obs_lwp,
+    'T_force_tend': obs_T_forcing, 'qv_force_tend': obs_q_forcing}
+
+  obs_fid.close()
+
+  return return_dict
+
+def read_arm_sgp_summer_1997_obs(obs_file, time_slices, date):
+  obs_time_slice_indices = []
+
+  obs_fid = Dataset(obs_file, 'r')
+
+  obs_year = obs_fid.variables['Year'][:]
+  obs_month = obs_fid.variables['Month'][:]
+  obs_day = obs_fid.variables['Day'][:]
+  #obs_hour = obs_fid.variables['hour'][:]
+  obs_time = obs_fid.variables['time_offset'][:]
+
+  #this file doesn't have the hour variable - calculate from the time offset (seconds from 00Z on 6/18/1997)
+  obs_hour = (((obs_time - 3)/3600.0)%24).astype(int)
+
+  obs_date = []
+  for i in range(obs_hour.size):
+      obs_date.append(datetime.datetime(obs_year[i], obs_month[i], obs_day[i], obs_hour[i], 0, 0, 0))
+  obs_date = np.array(obs_date)
+
+  for time_slice in time_slices:
+    start_date = datetime.datetime(time_slices[time_slice]['start'][0], time_slices[time_slice]['start'][1],time_slices[time_slice]['start'][2], time_slices[time_slice]['start'][3], time_slices[time_slice]['start'][4])
+    end_date = datetime.datetime(time_slices[time_slice]['end'][0], time_slices[time_slice]['end'][1],time_slices[time_slice]['end'][2], time_slices[time_slice]['end'][3], time_slices[time_slice]['end'][4])
+    start_date_index = np.where(obs_date == start_date)[0][0]
+    end_date_index = np.where(obs_date == end_date)[0][0]
+    obs_time_slice_indices.append([start_date_index, end_date_index])
+    #print start_date, end_date, start_date_index, end_date_index, obs_date[start_date_index], obs_date[end_date_index]
+
+  #find the index corresponding to the start of the simulations
+  obs_start_index = np.where(obs_date == date[0][0])[0]
+  obs_time = obs_time - obs_time[obs_start_index]
+
+
+  obs_pres_l = np.flipud(obs_fid.variables['lev'][:])*100.0 #pressure levels in mb
+
+  obs_cld = np.fliplr(obs_fid.variables['ARSCL_Cld'][:,:,0,0])/100.0
+  obs_T = np.fliplr(obs_fid.variables['Temp'][:,:,0,0])
+  obs_q = np.fliplr(obs_fid.variables['H2O_Mixing_Ratio'][:,:,0,0]/1000.0)
+  obs_u = np.fliplr(obs_fid.variables['u_wind'][:,:,0,0])
+  obs_v = np.fliplr(obs_fid.variables['v_wind'][:,:,0,0])
+  obs_precip = obs_fid.variables['Prec'][:,0,0]
+  # obs_shf = obs_fid.variables['SH'][:]
```

Comment:
We should avoid having large blocks of commented code. If this is code that is needed for debugging or other purposes can we handle that with an `if` statement rather than forcing the manual editing of code?